src/core/lib/event_engine/posix_engine/posix_endpoint.cc

```diff
@@ -1167,11 +1207,22 @@ PosixEndpointImpl::PosixEndpointImpl(EventHandle* handle,
   PosixSocketWrapper sock(handle->WrappedFd());
   fd_ = handle_->WrappedFd();
   GPR_ASSERT(options.resource_quota != nullptr);
+  auto peer_addr_string = sock.PeerAddressString();
   memory_owner_ = options.resource_quota->memory_quota()->CreateMemoryOwner(
-      *sock.PeerAddressString());
+      peer_addr_string.ok() ? *peer_addr_string : "");
   self_reservation_ = memory_owner_.MakeReservation(sizeof(PosixEndpointImpl));
-  local_address_ = *sock.LocalAddress();
-  peer_address_ = *sock.PeerAddress();
+  auto local_address = sock.LocalAddress();
+  if (local_address.ok()) {
+    local_address_ = *local_address;
+  } else {
+    local_address_ = EventEngine::ResolvedAddress{};
```

Comment:
`local_address_` should be default-constructed already.

Reply:
Fixed.